From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Todd Kjos <tkjos@google.com>
Date: Thu, 12 Dec 2019 11:36:29 -0800
Subject: ANDROID: README.md: add text on upstreaming patches

Added text to README.md to remind developers that
upstream maintainers need to see uses of exported
symbols if new exports are being added and that
enablement of out-of-tree functionality is not
a sufficient argument for patches to be added.

Change-Id: I633307e7633aab81cb8ae79a9a10bd2d4bc1cebb
Signed-off-by: Todd Kjos <tkjos@google.com>
---
 README.md | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/README.md b/README.md
index 110e8535b111..e10eee47488e 100644
--- a/README.md
+++ b/README.md
@@ -3,6 +3,12 @@
 1. BEST: Make all of your changes to upstream Linux. If appropriate, backport to the stable releases.
    These patches will be merged automatically in the corresponding common kernels. If the patch is already
    in upstream Linux, post a backport of the patch that conforms to the patch requirements below.
+   - Do not send patches upstream that contain only symbol exports. To be considered for upstream Linux,
+additions of `EXPORT_SYMBOL_GPL()` require an in-tree modular driver that uses the symbol -- so include
+the new driver or changes to an existing driver in the same patchset as the export.
+   - When sending patches upstream, the commit message must contain a clear case for why the patch
+is needed and beneficial to the community. Enabling out-of-tree drivers or functionality is not
+not a persuasive case.
 
 2. LESS GOOD: Develop your patches out-of-tree (from an upstream Linux point-of-view). Unless these are
    fixing an Android-specific bug, these are very unlikely to be accepted unless they have been
